packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+        connectedContainers = [];
+        closedContainers = [];
+        testNodes = new Map();
+        testNodes.set(mainDataStore.id, new TestNode(mainDataStore.id));
+        unrefDataStores = new Set();
+        refDataStores = new Set();
+        rootDataStores = new Set();
+        refDataStores.add(mainDataStore.id);
+        rootDataStores.add(mainDataStore.id);
+
+        await provider.ensureSynchronized();
+    });
+
+    afterEach(() => {
+
+    });
+
+    itExpects.skip("GC is notified when datastores are aliased.", [
+        {
+            eventName: "fluid:telemetry:Container:ContainerClose",
+            errorType: ContainerErrorType.clientSessionExpiredError,
+        },
+    ], async () => {
+        const mainContainerDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        mainContainerDataStore._root.set("test", "123");
+        mainContainerDataStore._root.set("123", "123");
+        connectedContainers.push(mainContainer);
+        await provider.ensureSynchronized();
+        const summarizeResults = mainContainerDataStore.containerRuntime.summarizeOnDemand({
+            reason: "GC Testing",
+        });
+
+        const summarySubmitted = await summarizeResults.summarySubmitted;
+        assert(summarySubmitted.success === true, "Expected successful summarizeOnDemand!");
+        assert(summarySubmitted.data !== undefined, "Expected summarizeOnDemand data!");
+        assert(summarySubmitted.data.stage === "submit", "Should have been submitted");
+        assert(summarySubmitted.data.summaryTree !== undefined, "summary tree should exist");
+        const summaryTree = summarySubmitted.data.summaryTree;
+        uploadedSummaries.push(summaryTree);
+
+        await sleep(defaultSessionExpiryDurationMs);
+
+        await loadContainer();
+        assert(mainContainer.closed === true, "Expected mainContainer to be closed!");
+        assert(uploadedSummaries.length > 0, `${uploadedSummaries.length}`);
+    });
+
+    const randomInteger = (min: number, max: number): number => {
+        return makeRandom().integer(min, max);
+    };
+
+    const loadNewContainer = async () => {
+        const newContainer = await loadContainer();
+        newContainer.on("closed", () => {
+            const index = connectedContainers.indexOf(newContainer);
+            assert(index >= 0, "Expected container to have been added to connectedContainers");
+            closedContainers.push(connectedContainers[index]);
+            connectedContainers.splice(index, 1);
+        });
+        connectedContainers.push(newContainer);
+    };
+
+    const closeRandomContainer = async () => {
+        if (connectedContainers.length <= 0) {
+            return;
+        }
+        const index = randomInteger(0, connectedContainers.length - 1);
+        return new Promise((resolve) => resolve(connectedContainers[index].close()));
```

Comment:
Do you mean to type this?

`return Promise.resolve().then(() => connectedContainers[index].close());`

As-is, it closes immediately (and synchronously), then returns basically the same as `Promise.resolve()` which is of no value that I can see.  Do the above if you want to queue a microtask to do the close.  If you don't need to do that, then drop all promise stuff from this function, it can be synchronous.

Reply:
Probably need to queue a microtask, the only thing is that I wanted this to return a promise so that I could have a list of promises to fire off.

Reply:
This function returns `Promise<void>`:  `const foo = async() => {}`

In other words, you don't need to explicitly create a promise in a case like that; simply declaring the function as async ensures that it returns a Promise.

(The converse isn't true:  You can have a non-async function that returns a Promise, it just won't necessarily act like you expect especially if an error is thrown. There's a lint rule to discourage this)